# 01 — Primeiro contato com os dados

**Objetivo:** carregar a base de fraude do MED (`EstatisticasFraudesPix`),
entender dimensões e tipos, converter a coluna de mês para data e checar se a
série temporal está completa — sem lacunas e sem duplicatas.

Faz parte da [etapa 2](../README.md#progresso) do projeto.

In [1]:
import pandas as pd

URL = "https://raw.githubusercontent.com/ivadivad/med-em-numeros/main/dados/fraude.csv"
df = pd.read_csv(URL)
df.shape

(52, 24)

## Dimensões e tipos

52 linhas — uma por mês, de jan/2022 a abr/2026 — e 24 colunas. Vale olhar os
tipos antes de qualquer conta: tudo que é "Qtde"/"Quantidade" veio como
inteiro, tudo que é "Valor" ou percentual veio como ponto flutuante. Nenhuma
coluna de texto, nenhum nulo.

In [2]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 52 entries, 0 to 51
Data columns (total 24 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   AnoMes                                        52 non-null     int64  
 1   QtdePixcontestados                            52 non-null     int64  
 2   Qtdecontestacoesaceitas                       52 non-null     int64  
 3   Qtdecontestacoesrejeitadas                    52 non-null     int64  
 4   Qtdecontestacoesaceitasacada100mil            52 non-null     float64
 5   QtdeUsuarioscommarcacoesdefraude              52 non-null     int64  
 6   QtdeChavesPixcommarcacoesdefraude             52 non-null     int64  
 7   ValorPixcontestadosaceitos                    52 non-null     float64
 8   QuantidadedevolvidaintegralmentepormeiodoMED  52 non-null     int64  
 9   ValorPixdevolvidosintegralmente               52 non-null     float64
 10  Qua

In [3]:
df.describe()

,AnoMes,QtdePixcontestados,Qtdecontestacoesaceitas,Qtdecontestacoesrejeitadas,Qtdecontestacoesaceitasacada100mil,QtdeUsuarioscommarcacoesdefraude,QtdeChavesPixcommarcacoesdefraude,ValorPixcontestadosaceitos,QuantidadedevolvidaintegralmentepormeiodoMED,ValorPixdevolvidosintegralmente,...,ValorPixnaodevolvidossaldoinsuficiente,Quantidadedenaodevolvidoscontaencerrada,Valornaodevolvidoscontaencerrada,Quantidadedenaodevolvidosmotivosdiversos,ValorPixnaodevolvidosmotivosdiversos,PercentualdeDevolucao,QtdePixbloqueadoscautelarmenteeliberados,ValorPixbloqueadoscautelarmenteeliberados,QtdePixbloqueadoscautelarmenteedevolvidos,ValorPixbloqueadoscautelarmenteedevolvidos
count,52.000000,5.200000e+01,52.000000,5.200000e+01,52.000000,52.000000,52.000000,5.200000e+01,52.000000,5.200000e+01,...,5.200000e+01,52.000000,5.200000e+01,52.000000,5.200000e+01,52.000000,5.200000e+01,5.200000e+01,52.000000,5.200000e+01
mean,202375.423077,1.021337e+06,271699.961538,7.496366e+05,7.027635,71731.538462,78834.576923,4.714933e+08,44099.115385,2.807686e+07,...,2.647900e+08,9581.384615,1.158846e+07,12968.769231,3.207342e+07,8.720385,8.421811e+05,2.322962e+09,117806.192308,5.921809e+07
std,127.111244,9.898714e+05,115834.584108,9.390247e+05,1.686219,29039.978281,31547.041037,1.970020e+08,29863.771744,1.647794e+07,...,1.212564e+08,8836.416451,8.863799e+06,4938.152848,1.981695e+07,2.475903,5.309931e+05,1.317716e+09,136491.717953,4.725657e+07
min,202201.000000,7.150800e+04,57777.000000,1.373100e+04,4.236000,20839.000000,22973.000000,1.225589e+08,5301.000000,4.261762e+06,...,6.207817e+07,2359.000000,3.313131e+06,3401.000000,7.775309e+06,2.860000,2.646000e+04,1.791425e+07,13319.000000,1.004904e+07
25%,202301.750000,2.614872e+05,173667.750000,8.578225e+04,5.633750,47020.500000,52021.750000,3.034688e+08,17305.750000,1.540068e+07,...,1.574045e+08,4631.250000,5.964543e+06,9309.250000,1.998561e+07,7.462500,5.247292e+05,1.794700e+09,18568.250000,2.026848e+07
50%,202402.500000,7.689095e+05,284174.500000,4.414275e+05,7.117000,72184.000000,80355.500000,4.662696e+08,43308.500000,2.628086e+07,...,2.836248e+08,6723.500000,9.022878e+06,12417.000000,2.674059e+07,8.530000,8.159345e+05,2.267469e+09,38438.500000,3.812990e+07
75%,202503.250000,1.279282e+06,346450.750000,9.525460e+05,7.801500,97690.750000,107544.750000,6.150933e+08,61983.750000,3.615707e+07,...,3.367626e+08,9380.500000,1.283358e+07,15667.750000,3.425149e+07,9.570000,1.124247e+06,3.285287e+09,158254.000000,9.275700e+07
max,202604.000000,3.771626e+06,530753.000000,3.449104e+06,11.088000,130576.000000,138420.000000,8.598451e+08,122792.000000,7.463535e+07,...,6.243183e+08,43379.000000,4.719148e+07,26401.000000,1.060507e+08,16.230000,2.142208e+06,5.485195e+09,573308.000000,1.842573e+08


In [4]:
df.head()

,AnoMes,QtdePixcontestados,Qtdecontestacoesaceitas,Qtdecontestacoesrejeitadas,Qtdecontestacoesaceitasacada100mil,QtdeUsuarioscommarcacoesdefraude,QtdeChavesPixcommarcacoesdefraude,ValorPixcontestadosaceitos,QuantidadedevolvidaintegralmentepormeiodoMED,ValorPixdevolvidosintegralmente,...,ValorPixnaodevolvidossaldoinsuficiente,Quantidadedenaodevolvidoscontaencerrada,Valornaodevolvidoscontaencerrada,Quantidadedenaodevolvidosmotivosdiversos,ValorPixnaodevolvidosmotivosdiversos,PercentualdeDevolucao,QtdePixbloqueadoscautelarmenteeliberados,ValorPixbloqueadoscautelarmenteeliberados,QtdePixbloqueadoscautelarmenteedevolvidos,ValorPixbloqueadoscautelarmenteedevolvidos
0,202507,1328139,325959,1002180,5.435,106481,117217,6.985666e+08,57479,40389809.35,...,3.458664e+08,6961,12980548.19,16525,53022123.99,9.92,1674847,1.950942e+09,104320,6.974826e+07
1,202408,1146697,517930,628767,10.339,97290,110038,8.418846e+08,100372,51285171.91,...,4.624505e+08,19518,28952813.32,16115,75168786.32,9.87,1123852,1.782125e+09,183984,7.504241e+07
2,202410,1262997,494691,768306,9.707,103507,116606,6.285789e+08,122792,39914254.85,...,3.875005e+08,8646,7460872.65,13725,29491240.83,8.74,1378029,1.838606e+09,166090,1.106953e+08
3,202405,841711,346370,495341,7.496,78882,87448,5.373407e+08,51410,29102491.12,...,3.156256e+08,7460,9979213.15,20565,54457360.78,8.83,1073109,4.119718e+09,70283,1.113718e+08
4,202512,3771626,322522,3449104,4.543,105158,110824,5.738475e+08,52795,36932178.49,...,3.030013e+08,8235,8299089.51,11307,29654498.64,9.60,530773,3.082296e+09,354882,8.082697e+07


**Leitura do que apareceu acima:**

- `df.info()` confirma: 52 linhas, 24 colunas, nenhum valor nulo em nenhuma
  coluna (todas com `52 non-null`).
- `df.describe()` mostra a escala bem diferente entre colunas — contagens na
  casa dos milhares a milhões, valores em reais na casa das centenas de
  milhões (`ValorPixcontestadosaceitos` vai de ~R$123 milhões a ~R$860
  milhões por mês), e `PercentualdeDevolucao` sempre entre 0 e 100 (no caso,
  entre 2,86 e 16,23). Mistura escalas assim importa pra quando for plotar
  mais de uma série no mesmo gráfico (etapa 5 em diante).
- `df.head()` mostra as primeiras linhas — e não dá pra assumir que o
  CSV vem em ordem cronológica: na versão do coletor que gerou esta base,
  não vinha (efeito de como a Olinda responde; ver `montar_url` e
  `consolidar` em `coletar.py`). Isso não é problema de dado — é só
  ordenar explicitamente antes de qualquer análise de série temporal, o
  que a próxima seção já faz.

## Convertendo `AnoMes` para data

`AnoMes` chega como inteiro no formato `AAAAMM` (ex: `202201` = janeiro de
2022). Pra usar como eixo de tempo depois, converto pra `datetime` com
`pd.to_datetime(..., format="%Y%m")` e já ordeno por essa coluna.

In [5]:
df["Data"] = pd.to_datetime(df["AnoMes"], format="%Y%m")
df = df.sort_values("Data").reset_index(drop=True)
df[["AnoMes", "Data"]].head()

,AnoMes,Data
0,202201,2022-01-01
1,202202,2022-02-01
2,202203,2022-03-01
3,202204,2022-04-01
4,202205,2022-05-01


## Lacunas e duplicatas

Duas checagens antes de confiar na série: (1) nenhum mês repetido, (2)
nenhum mês faltando entre o primeiro e o último.

In [6]:
duplicados = df["AnoMes"].duplicated().sum()
print(f"meses duplicados: {duplicados}")

meses duplicados: 0


In [7]:
esperado = pd.date_range(df["Data"].min(), df["Data"].max(), freq="MS")
faltando = sorted(set(esperado) - set(df["Data"]))

print(f"intervalo: {df['Data'].min():%Y-%m} a {df['Data'].max():%Y-%m}")
print(f"meses esperados: {len(esperado)}  |  meses presentes: {len(df)}")
print(f"meses faltando: {len(faltando)}")
faltando

intervalo: 2022-01 a 2026-04
meses esperados: 52  |  meses presentes: 52
meses faltando: 0


[]

**Resultado: 0 meses duplicados, 0 meses faltando.** A série vai de janeiro
de 2022 a abril de 2026, 52 meses seguidos, uma linha por mês — exatamente o
que o `fraude_cobertura.csv` do coletor já indicava, agora confirmado a
partir do CSV consolidado mesmo, sem depender do relatório da coleta.

Base validada na dimensão que importa pra esta etapa (completude). A validação
de que os *números* fazem sentido — reconstruir `PercentualdeDevolucao` e
bater com uma fonte externa — é a [etapa 3](../README.md#progresso).